subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -656,6 +656,11 @@ static int mod_sub_status(struct bt_mesh_model *model,
 
 	mod_id = net_buf_simple_pull_le16(buf);
 
+	if (cli->cb && cli->cb->mod_sub_status) {
```

Comment:
What is the reason to invoke callback before context matching not after?
Will it cause a situation when the application gets a notification about a status that, actually, doesn't impact anything?